modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientKeyValueBinaryView.java

```diff
@@ -111,26 +112,26 @@ public CompletableFuture<Map<Tuple, Tuple>> getAllAsync(@Nullable Transaction tx
             return emptyMapCompletedFuture();
         }
 
-        BiFunction<Collection<Tuple>, PartitionAwarenessProvider, CompletableFuture<Map<Tuple, Tuple>>> clo = (batch, provider) -> {
+        List<Transaction> txns = new ArrayList<>();
+
+        MapFunction<Tuple, Map<Tuple, Tuple>> clo = (batch, provider, txRequired) -> {
+            Transaction tx0 = tbl.startTxIfNeeded(tx, txns, txRequired);
```

Comment:
It so unclear to maintain an array list (txns) in the other method instead of doing it in this place.
```
Transaction tx0 = tbl.startTxIfNeeded(tx, txRequired);
txns.add(tx0);
```

Reply:
Or even better, add a method (getTransactions()) to MapFunction.